src/app/auth/auth-config.module.ts

```diff
@@ -0,0 +1,38 @@
+import { NgModule } from '@angular/core';
+import { AuthModule, LogLevel, OpenIdConfiguration } from 'angular-auth-oidc-client';
+import { HTTP_INTERCEPTORS } from '@angular/common/http';
+import { WharfAuthInterceptor } from './wharf-auth.interceptor';
+
+
+@NgModule({
+  imports: [
+    AuthModule.forRoot({
+      config: AuthConfigModule.defaultConfig,
+    }),
+  ],
+  exports: [AuthModule],
+  declarations: [],
+  providers: [
+    {provide: HTTP_INTERCEPTORS, useClass: WharfAuthInterceptor, multi: true},
+  ],
+})
+export class AuthConfigModule {
+  static defaultConfig: OpenIdConfiguration = {
+    authority: 'https://login.microsoftonline.com/841df554-ef9d-48b1-bc6e-44cf8543a8fc' +
+      '/v2.0/.well-known/openid-configuration',
+    redirectUrl: window.location.origin,
+    postLogoutRedirectUri: window.location.origin,
+    clientId: '01fcb3dc-7a2b-4b1c-a7d6-d7033089c779', // Application (client) ID
+    scope: 'openid profile email offline_access api://wharf-internal/read api://wharf-internal/admin api://wharf-internal/deploy',
+    responseType: 'id_token token',
+    ignoreNonceAfterRefresh: true,
+    silentRenew: true,
+    useRefreshToken: true,
+    logLevel: LogLevel.Error,
+    maxIdTokenIatOffsetAllowedInSeconds: 600,
+    issValidationOff: false,
+    autoUserInfo: false,
+  };
```

Comment:
Trying to summarize what we talked about in one of our meetings, @fredx30 please correct me on the actual state of this. At least what I want to see in this PR before we can release it is:

- Make these configs here in `src/app/auth/auth-config.module.ts` to originate from `src/assets/config.json`. Even if these configs are unsensitive/non-confidential, I will not accept us having them hardcoded for our official releases. An unaltered Docker image should not have any OIDC configured.

- We only need wharf-web to require login if the auth is enabled, otherwise it's fully OK if this code is loaded but completely unused when the auth is disabled.

I need to take a closer look at this code, but want to wait to not add any frivolent review comments that are instantly outdated by your changes.

Reply:
Old discussion, dont remember. Resolving for now.